lib/worker.js

```diff
@@ -182,6 +184,7 @@ worker.prototype.perform = function(job, callback) {
                   if(returnCounter !== 3){
                     self.emit('failure', self.queue, job, callbackError);
                   }else{
+                    self.result = result;
```

Comment:
If you are saving the result to `self`, you shouldn't also pass it via callback... that would get confusing.